packages/workflow/src/internals.ts

```diff
@@ -577,6 +686,34 @@ export class Activator implements ActivationHandler {
     }
   }
 
+  public dispatchBufferedUpdates(): void {
+    const bufferedUpdates = this.bufferedUpdates;
+    while (bufferedUpdates.length) {
+      const foundIndex = bufferedUpdates.findIndex((update) => update.name && this.updateHandlers.has(update.name));
+      if (foundIndex === -1) {
+        // No buffered Updates have a handler yet.
+        break;
+      }
+      const [update] = bufferedUpdates.splice(foundIndex, 1);
+      this.doUpdate(update);
+    }
+  }
+
+  public rejectBufferedUpdates(): void {
+    while (this.bufferedUpdates.length) {
+      const update = this.bufferedUpdates.shift();
```

Comment:
```suggestion
    const bufferedUpdates = this.bufferedUpdates;
    this.bufferedUpdates = [];
    for (const update of bufferedUpdates) {
```

Reply:
And you could also just skip the creation of a new array if there aren't any buffered updates.